engine/src/main/java/org/terasology/logic/health/DoDestroyEvent.java

```diff
@@ -36,11 +38,36 @@ public EntityRef getInstigator() {
         return instigator;
     }
 
+    public String getInstigatorString() {
+        if (instigator.getParentPrefab() != null) {
```

Comment:
If the instigator was a player, this will always show "Player". I would like to see the actual player name here.
For the kill command, this does not work because `HealthCommands` adds only `clientComp.character` instead of `clientComp.clientInfo` (clientInfo is an Entity with the `DisplayNameComponent`, character does not contain a display name). Can you check if it makes more sense to pass the client info instead? Another option would be to duplicate the DisplayNameComponent to the character entity but this place may need some more investigation :)

It would also be nice to support a general DisplayNameComponent, the prefab name may not always be human readable (thinking of something like `SmeltingCapableCampFire`). Keeping the logic with the name parsing as fallback is still okay.